src/BridgedRoom.ts

```diff
@@ -206,10 +224,30 @@ export class BridgedRoom {
         this.slackBotId =  opts.slack_bot_id;
         this.accessToken = opts.access_token;
         this.accessScopes = opts.access_scopes;
+        this.slackType = opts.slack_type || "channel";
+        if (opts.is_private === undefined) {
+            opts.is_private = false;
+        }
+        this.isPrivate = opts.is_private;
+        this.puppetOwner = opts.puppet_owner;
 
         this.dirty = true;
     }
 
+    public updateUsingChannelInfo(channelInfo: ConversationsInfoResponse) {
+        const chan = channelInfo.channel;
+        this.setValue("isPrivate", chan.is_private);
+        if (chan.is_channel) {
+            this.setValue("slackType", "channel");
+        } else if (chan.is_group) {
+            this.setValue("slackType", "group");
+        } else if (chan.is_im) {
```

Comment:
we can probably give `is_im` a better name

Reply:
This is a slack response, so this func is just handling it.